core/shared/src/main/scala/org/http4s/headers/Cookie.scala

```diff
@@ -28,23 +28,24 @@ object Cookie {
   def apply(head: RequestCookie, tail: RequestCookie*): `Cookie` =
     apply(NonEmptyList(head, tail.toList))
 
-  def parse(s: String): ParseResult[Cookie] =
-    ParseResult.fromParser(parser, "Invalid Cookie header")(s)
+  def parse(s: String): ParseResult[Cookie] = {
```

Comment:
The logic seems right, but It's new to have this method silently discard errors when none of our other parsers behave this way.  I think it would be more principled if we:
1. Had a `parseWithWarnings(headerValue: String): Ior[NonEmptyList[ParseFailure], A]` on the `Header` typeclass.
2. For binary compatibility, gave it a default that delegates to `parse`.
3. Integrate it into the `Headers.getWithWarnings` that already exists.
4. Your method here is close to a good implementation of that `parseWithWarnings`.

I guess the real question is what relationship is expected between `get` and `getWithWarnings` when the `Ior` returns both.  My gut feeling is warnings are "fatal" and we return a Left, and this PR suggests warnings are "non-fatal" and we should return a best effort.  I'm curious what others think.